apps/yggdrasil_web/lib/guardian_serializer.ex

```diff
@@ -1,12 +1,20 @@
 defmodule YggdrasilWeb.GuardianSerializer do
   @behaviour Guardian.Serializer
 
-  alias Yggdrasil.Repo
-  alias Yggdrasil.User
+  require Ecto.Query
+  import Ecto.Query, only: [from: 1, from: 2]
+
+  alias Yggdrasil.{Repo, User, Role}
 
   def for_token(user = %User{}), do: {:ok, "user:#{user.id}"}
   def for_token(_), do: {:error, "Unknown resource type."}
 
-  def from_token("user:" <> user_id), do: {:ok, Repo.get(User, user_id)}
+  def from_token("user:" <> user_id) do
```

Comment:
This gets called when the plug the Guardian plug loads the user, which I think is only if it's not loaded so this should only be once per connection. So as long as the connection is alive it's stored there.